// rebrowser-patches/eslint.config.cjs
const config= {
  parserOptions: {
    ecmaVersion: 'latest',
    sourceType: 'module',
  },
  extends: [

  ],
  plugins: [
    'unused-imports',
    'simple-import-sort',
    'import',
  ],
  rules: {
    'no-console': 'warn',
    'quote-props': ['error', 'as-needed'],
    semi: ['error', 'never'],
    indent: ['error', 2],
    'spaced-comment': ['error', 'always'],
    'object-curly-newline': [
      'error', {
        consistent: true,
        multiline: true,
        minProperties: 3,
      },
    ],
    'object-curly-spacing': ['error', 'always'],
    'object-property-newline': ['error', { allowAllPropertiesOnSameLine: false }],
    'object-shorthand': ['error', 'always'],
    'comma-dangle': ['error', 'always-multiline'],
    'comma-spacing': [
      'error', {
        before: false,
        after: true,
      },
    ],
    'array-bracket-newline': ['error', { multiline: true }],
    'array-element-newline': [
      'error', {
        ArrayExpression: 'consistent',
        ArrayPattern: { minItems: 3 },
      },
    ],
    'array-bracket-spacing': 'error',
    'arrow-spacing': [
      'error', {
        before: true,
        after: true,
      },
    ],
    'keyword-spacing': 'error',
    'key-spacing': 'error',
    'space-before-blocks': ['error', 'always'],
    'space-before-function-paren': [
      'error', {
        anonymous: 'always',
        named: 'never',
        asyncArrow: 'always',
      },
    ],
    'space-infix-ops': ['error', { int32Hint: true }],
    'space-in-parens': ['error', 'never'],
    'switch-colon-spacing': 'error',
    quotes: ['error', 'single'],
    'no-multi-spaces': 'error',
    'no-unused-vars': 'off',
    'unused-imports/no-unused-imports': 'error',
    'unused-imports/no-unused-vars': [
      'warn', {
        vars: 'all',
        varsIgnorePattern: '^_',
        args: 'after-used',
        argsIgnorePattern: '^_',
      },
    ],
    'import/extensions': [
      'error', 'ignorePackages', {
        js: 'always',
        jsx: 'always',
      },
    ],
    'simple-import-sort/imports': 'error',
    'simple-import-sort/exports': 'error',
    'import/first': 'error',
    'import/newline-after-import': 'error',
    'import/no-duplicates': 'error',
    'no-trailing-spaces': ['error'],
    'no-multiple-empty-lines': ['error', { max: 1 }],
    'eol-last': ['error', 'always'],
    'jsx-quotes': ['error', 'prefer-double'],
    'prefer-const': ['error'],
  },
}

module.exports = config
